src/codecs/resize/options.tsx

```diff
@@ -0,0 +1,128 @@
+import { h, Component } from 'preact';
+import linkState from 'linkstate';
+import { bind, inputFieldValueAsNumber } from '../../lib/util';
+import { ResizeOptions } from './resize';
+
+interface Props {
+  options: ResizeOptions;
+  aspect: number;
+  onChange(newOptions: ResizeOptions): void;
+}
+
+interface State {
+  maintainAspect: boolean;
+}
+
+export default class QuantizerOptions extends Component<Props, State> {
+  state: State = {
+    maintainAspect: true,
+  };
+
+  form: HTMLFormElement = undefined as any as HTMLFormElement;
```

Comment:
Interested in thought on this. Technically `form` isn't defined until first mount, but it's a pain adding throws to every method.

Is this a reasonable cheat, or would it be better to create a decorator that throws if form is missing?

Reply:
If we decide this is ok, I should add a comment explaining it.